WordPress/Classes/ViewRelated/Me/App Settings/Boolean User Defaults/BooleanUserDefaultsDebugViewModel.swift

```diff
@@ -0,0 +1,104 @@
+                key.localizedCaseInsensitiveContains(query) || userDefault.title.localizedCaseInsensitiveContains(query)
+            }
+            if sectionKey.localizedCaseInsensitiveContains(query) || !filteredUserDefaults.isEmpty {
+                filteredSections[sectionKey] = filteredUserDefaults.isEmpty ? userDefaults : filteredUserDefaults
+            }
+        }
+        return filteredSections
+    }
+
+    init(coreDataStack: CoreDataStack = ContextManager.shared,
+         persistentRepository: UserPersistentRepository = UserPersistentStoreFactory.instance()) {
+        self.coreDataStack = coreDataStack
+        self.persistentRepository = persistentRepository
+        load()
+    }
+
+    func load() {
+        let allUserDefaults = persistentRepository.dictionaryRepresentation()
+        var loadedUserDefaultsSections = BooleanUserDefaultsSections()
+        var otherSection = BooleanUserDefaultEntries()
+
+        for (entryKey, entryValue) in allUserDefaults {
+            if let groupedUserDefaults = entryValue as? [String: Bool], !isFeatureFlagsSection(entryKey) {
+                loadedUserDefaultsSections[entryKey] = processGroupedUserDefaults(groupedUserDefaults)
+            } else if let booleanUserDefault = entryValue as? Bool, !isGutenbergUserDefault(entryKey) {
+                otherSection[entryKey] = BooleanUserDefault(title: entryKey, value: booleanUserDefault)
+            }
+        }
+        if !otherSection.isEmpty {
+            loadedUserDefaultsSections[Strings.otherBooleanUserDefaultsSectionID] = otherSection
+        }
+        allUserDefaultsSections = loadedUserDefaultsSections
+    }
+
+    private func processGroupedUserDefaults(_ userDefaults: [String: Bool]) -> BooleanUserDefaultEntries {
+        userDefaults.reduce(into: BooleanUserDefaultEntries()) { result, keyValue in
+            let (key, value) = keyValue
+            result[key] = processSingleUserDefault(key: key, value: value)
+        }
+    }
+
+    private func processSingleUserDefault(key: String, value: Bool) -> BooleanUserDefault {
+        if let siteID = Int(key), let blogURL = try? Blog.lookup(withID: siteID, in: coreDataStack.mainContext)?.url {
+            return BooleanUserDefault(title: blogURL, value: value)
+        } else {
+            return BooleanUserDefault(title: key, value: value)
+        }
+    }
+
+    func updateUserDefault(_ value: Bool, forSection sectionID: String, forUserDefault userDefaultID: String) {
+        if sectionID == Strings.otherBooleanUserDefaultsSectionID {
+            persistentRepository.set(value, forKey: userDefaultID)
+        } else if var section = allUserDefaultsSections[sectionID] {
+            section[userDefaultID] = BooleanUserDefault(title: userDefaultID, value: value)
+            let sectionValues = section.mapValues { $0.value }
+            persistentRepository.set(sectionValues, forKey: sectionID)
+        }
+        load()
+    }
```

Comment:
[Suggestion] I think reloading the whole list for every toggle switch is a little bit overkill. Ideally, we want to change only the row associated with the user default entry. I think you can achieve that by changing the type `BooleanUserDefault` to: 

```swift
final class BooleanUserDefault: ObservableObject {
    let title: String
    @Published var value: Bool

    init(title: String, value: Bool) {
        self.title = title
        self.value = value
    }
}
```

Then you can bind the `value` with the UI.

Reply:
Sounds absolutely reasonable to me! Thanks for catching this, I'll improve the performance of this.